api/openapi.yaml

```diff
@@ -0,0 +1,209 @@
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Application"
+        "400":
+          description: Invalid address supplied
+        "404":
+          description: Application not found
+    put:
+      tags:
+        - applications
+      summary: Adds an application
+      description: Adds an application by its address
+      operationId: addApplication
+      parameters:
+        - name: address
+          in: path
+          description: Application address
+          required: true
+          schema:
+            type: string
+            format: address
+            example: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+            pattern: "^0x([0-9a-fA-F]{40})$"
+      requestBody:
+        $ref: "#/components/requestBodies/Application"
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Application"
+    delete:
+      tags:
+        - applications
+      summary: Deletes an application
+      description: delete an application
+      operationId: deleteApplication
+      parameters:
+        - name: address
+          in: path
+          description: Application address to delete
+          required: true
+          schema:
+            type: string
+            format: address
+            example: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+            pattern: "^0x([0-9a-fA-F]{40})$"
+      responses:
+        "202":
+          description: Delete request accepted
+        "400":
+          description: Invalid address value
+        "404":
+          description: Application not found
+components:
+  schemas:
+    PaginatedResult:
+      type: object
+      properties:
+        total_count:
+          type: integer
+        offset:
+          type: integer
+        limit:
+          type: integer
+        data:
+          type: array
+          items: {}
+      required:
+        - total_count
+        - offset
+        - limit
+        - data
+    Application:
+      required:
+        - address
+        - blockNumber
+        - templateHash
+        - snapshotUrl
+      type: object
+      properties:
+        address:
+          type: string
+          format: address
+          example: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+          pattern: "^0x([0-9a-fA-F]{40})$"
+        blockNumber:
+          type: integer
+          example: 456311
+        templateHash:
+          type: string
+          format: hash
+          pattern: "^([0-9a-fA-F]{64})$"
+          example: "9a8d6ce861c71b592c236013b5c3fa167501431def4ad86ff59a3b12aa331dfc"
+        snapshotUrl:
+          type: string
+          format: url
+        status:
+          type: string
+          description: application status
+          enum:
+            - started
+            - starting
+            - stopping
+            - stopped
```

Comment:
When do these two cases happen? Will we keep the applications as stopped once we delete them?

Reply:
I can remove the "stopped", but leave the "stopping".

I think there will be the case where a client will issue a DELETE, and poll for the completion of the operation (as DELETE returns 202). Then it will polling waiting for a 404 of the get. It's kind of weird.

Reply:
Yes, this is weird. We can leave the stopped to avoid that.